src/main/java/apoc/periodic/Periodic.java

```diff
@@ -346,11 +348,15 @@ public Pattern regNoCaseMultiLine(String pattern) {
                         retried.addAndGet(retry(consumer,params,0,retries));
                     } catch (Exception e) {
                         failedOps.addAndGet(batchsize);
+                        if (failedParams >= 0) {
+                            failedParamsMap.put(Long.toString(finalBatches), batch.subList(0, Math.min(failedParams+1, batch.size())));
```

Comment:
Remember subList keeps a pointer to the original list. So it might be better to use new ArrayList(sublist)